src/main/java/org/isf/menu/gui/UserBrowsing.java

```diff
@@ -283,6 +265,18 @@ public void ancestorAdded(AncestorEvent event) {
 						model.fireTableDataChanged();
 						table.updateUI();
 					}
+				} catch (OHDataIntegrityViolationException ex) {
+					answer = MessageDialog.yesNo(null, "angal.userbrowser.softdeleteuser.fmt.msg", selectedUser.getUserName());
+					if (answer == JOptionPane.YES_OPTION) {
+						try {
+							selectedUser.setDeleted(true);
+							userBrowsingManager.updateUser(selectedUser);
+							model.fireTableDataChanged();
+							table.updateUI();
+						} catch (OHServiceException e) {
+							OHServiceExceptionUtil.showMessages(e);
+						}
+					}
```

Comment:
Given my proposed changes [here](https://github.com/informatici/openhospital-core/pull/1439/files#r1833373755) we could even avoid to call the manager again if we already know that the user is already soft-deleted.